table-data-gateway/README.md

```diff
@@ -0,0 +1,25 @@
+---  
+layout: pattern  
+title: Table Data Gateway
+folder: table-data-gateway 
+permalink: /patterns/table-data-gateway/  
+categories: Behavioral
+tags:
+- Database
+---  
+
+## Intent
+Table Data Gateway is a design pattern in which an object acts as a gateway to a database table. The idea is to separate the responsibility of fetching items from a database from the actual usages of those objects. Users of the gateway are then insulated from changes to the way objects are stored in the database.
```

Comment:
Add explanation section after intent. Please see the other patterns how it's supposed to be implemented e.g. https://github.com/iluwatar/java-design-patterns/tree/master/abstract-factory